src/commands/up.rs

```diff
@@ -143,85 +93,48 @@ impl UpCommand {
-                Some(TlsConfig {
-                    cert_path,
-                    key_path,
-                })
+        let manifest_url = match app.info.origin {
+            spin_manifest::ApplicationOrigin::File(path) => {
+                format!("file://{}", path.canonicalize()?.to_string_lossy())
+            }
+            spin_manifest::ApplicationOrigin::Bindle { id, server } => {
+                format!("bindle+{}?id={}", server, id)
             }
-            (None, None) => None,
-            _ => unreachable!(),
         };
 
-        let wasmtime_config = self.wasmtime_default_config()?;
-
-        let follow = self.follow_components();
-
-        match &app.info.trigger {
-            ApplicationTrigger::Http(_) => {
-                run_trigger(
-                    app,
-                    ExecutionOptions::<HttpTrigger>::new(
-                        self.opts.log.clone(),
-                        follow,
-                        HttpTriggerExecutionConfig::new(self.opts.address, tls),
-                    ),
-                    Some(wasmtime_config),
-                )
-                .await?;
-            }
-            ApplicationTrigger::Redis(_) => {
-                run_trigger(
-                    app,
-                    ExecutionOptions::<RedisTrigger>::new(self.opts.log.clone(), follow, ()),
-                    Some(wasmtime_config),
-                )
-                .await?;
-            }
+        let trigger_type = match app.info.trigger {
+            ApplicationTrigger::Http(_) => "http",
+            ApplicationTrigger::Redis(_) => "redis",
         };
 
-        // We need to be absolutely sure it stays alive until this point: we don't want
-        // any temp directory to be deleted prematurely.
-        drop(working_dir_holder);
+        let trigger_args = if self.help {
+            vec![OsString::from("--help")]
+        } else {
+            self.trigger_args
+        };
 
-        Ok(())
-    }
-    fn wasmtime_default_config(&self) -> Result<wasmtime::Config> {
-        let mut wasmtime_config = wasmtime::Config::default();
-        if !self.opts.disable_cache {
-            match &self.opts.cache {
-                Some(p) => wasmtime_config.cache_config_load(p)?,
-                None => wasmtime_config.cache_config_load_default()?,
-            };
+        // The docs for `current_exe` warn that this may be insecure because it could be executed
+        // via hard-link. I think it should be fine as long as we aren't `setuid`ing this binary.
+        let mut cmd = std::process::Command::new(std::env::current_exe().unwrap());
+        cmd.arg("trigger")
+            .env("SPIN_WORKING_DIR", working_dir)
+            .env("SPIN_MANIFEST_URL", manifest_url)
+            .env("SPIN_TRIGGER_TYPE", trigger_type)
+            .arg(trigger_type)
+            .args(trigger_args);
+
+        if let Some(bindle_server) = self.server {
+            cmd.env(BINDLE_URL_ENV, bindle_server);
```

Comment:
If you are passing it this way, do you need the server in the URL passed to the trigger executor?  Which one should a trigger treat as authoritative?

Reply:
It feels like we would need the information in both?

Reply:
This turns the clap field (which can be passed as an arg or env var) into always an env var. As above, I'm not sure this is a good long term solution, but its awkward to deduplicate without more refactoring.